### ISSUE TITLE
space-in-brackets doesn't account for parens

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#1517]
If I add these two test cases to the expected invalid code in `tests/lib/rules/space-in-brackets.js`, they don't actually produce any errors.

``` js
// ...
invalid: [
    // ...
    {
        code: "var foo = obj[(1 + 1)]",
        args: [2, "never", {singleValue: true}],
        errors: [
            {
                message: "A space is required after '['",
                type: "ArrayExpression"
            },
            {
                message: "A space is required before ']'",
                type: "ArrayExpression"
            }
        ]
    },
    {
        code: "var obj = {'foo': ([1, 2])};",
        args: [2, "never", {"arraysInObjects": true}],
        errors: [
            {
                message: "A space is required before '}'",
                type: "ObjectExpression"
            }
        ]
    },
    // ...
]
// ...
```

``` sh
$ node_modules/mocha/bin/mocha tests/lib/rules/space-in-brackets.js

  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․
  ․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․

  136 passing (144ms)
  2 failing

  1) space-in-brackets var foo = obj[(1 + 1)]:
     AssertionError: Should have 2 errors but had 0: []: expected 0 to equal 2


  2) space-in-brackets var obj = {'foo': ([1, 2])};:
     AssertionError: Should have 1 errors but had 0: []: expected 0 to equal 1
```
